src/Arcus.Observability.Telemetry.Serilog.Filters/TelemetryTypeFilter.cs

```diff
@@ -44,27 +48,36 @@ public static TelemetryTypeFilter On(TelemetryType telemetryType)
         /// </summary>
         /// <param name="telemetryType">Current telemetry type that should be filtered</param>
         /// <param name="isTrackingEnabled">Indication whether or not this telemetry type should be tracked</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="telemetryType"/> is outside the bounds of the enumeration.</exception>
         public static TelemetryTypeFilter On(TelemetryType telemetryType, bool isTrackingEnabled)
         {
+            Guard.For(() => !Enum.IsDefined(typeof(TelemetryType), telemetryType), 
+                new ArgumentOutOfRangeException(nameof(telemetryType), telemetryType, "Requires a type of telemetry that's within the supported value range of the enumeration"));
+            
             return new TelemetryTypeFilter(telemetryType, isTrackingEnabled);
         }
 
         /// <inheritdoc />
         public bool IsEnabled(LogEvent logEvent)
         {
-            if (IsTrackingEnabled == true)
+            if (IsTrackingEnabled is true)
```

Comment:
What's the advantage of writing `is true` instead of `== true` ?

Reply:
`is` forces the terms to be constants l, that's also why `... is null` is much safer than `... == null` bc the `==` can be overwritten. It's a personal choice maybe or my paranoia speaking. I tend to use `is` where possible.